File: client/src/main/java/com/example/client/MainActivity.java
package com.example.client;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import org.json.JSONException;
import org.json.JSONObject;

import java.io.BufferedInputStream;
import java.io.BufferedReader;
import java.io.ByteArrayOutputStream;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.io.OutputStream;
import java.io.PrintWriter;
import java.net.HttpURLConnection;
import java.net.URL;
import java.util.concurrent.ThreadPoolExecutor;

public class MainActivity extends AppCompatActivity {

    private ImageView img;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        img = findViewById(R.id.img);

        new Thread(
                new Runnable() {
                    @Override
                    public void run() {
                        String path = "https://api.weixin.qq.com/cgi-bin/token?grant_type=client_credential&appid=wxf47607581149e011&secret=cc7aa6caa5bd1406406b6b2c77c3088f";

                        String result = doGetHttp(path);
                        try {
                            JSONObject jsonObject = new JSONObject(result);
                            getCode(jsonObject.getString("access_token"));
                            Log.e("code--->", jsonObject.getString("access_token"));

                        } catch (JSONException e) {
                            e.printStackTrace();
                        }
                        Log.e("--------", result);
                    }
                }
        ).start();

    }

    public String doGetHttp(String path) {
        String result = "";
        try {
            URL url = new URL(path);
            HttpURLConnection connection = (HttpURLConnection) url.openConnection();
            connection.setRequestMethod("GET");
            connection.connect();
            StringBuffer buffer = new StringBuffer();
            if (connection.getResponseCode() == 200) {
                BufferedReader reader = new BufferedReader(new InputStreamReader(connection.getInputStream()));
                String str1;
                while ((str1 = reader.readLine()) != null) {
                    buffer.append(str1);
                }
                result = buffer.toString();
            } else {
                result = "网络异常";
            }
        } catch (Exception e) {
            Log.e("异常", e.toString());
            result = "网络异常";
        }
        return result;
    }


    public void getCode(String token) {
        try {
            URL url = new URL("https://api.weixin.qq.com/wxa/getwxacodeunlimit?access_token=" + token);
            HttpURLConnection httpURLConnection = (HttpURLConnection) url.openConnection();
            httpURLConnection.setRequestMethod("POST");// 提交模式
            // conn.setConnectTimeout(10000);//连接超时 单位毫秒
            // conn.setReadTimeout(2000);//读取超时 单位毫秒
            // 发送POST请求必须设置如下两行
            httpURLConnection.setDoOutput(true);
            httpURLConnection.setDoInput(true);
            // 获取URLConnection对象对应的输出流
            PrintWriter printWriter = new PrintWriter(httpURLConnection.getOutputStream());
            // 发送请求参数
            JSONObject paramJson = new JSONObject();
            //这就是你二维码里携带的参数 String型  名称不可变
            paramJson.put("scene", "133");
            //注意该接口传入的是page而不是path
//            paramJson.put("page", page);
            //这是设置扫描二维码后跳转的页面
            paramJson.put("width", 200);
            paramJson.put("is_hyaline", true);
            paramJson.put("auto_color", true);
            printWriter.write(paramJson.toString());
            // flush输出流的缓冲
            printWriter.flush();
            InputStream inputStream = httpURLConnection.getInputStream();
            ByteArrayOutputStream baos = new ByteArrayOutputStream();
            byte[] buffer = new byte[1024];
            int len = -1;
            while ((len = inputStream.read(buffer)) != -1) {
                baos.write(buffer, 0, len);
            }
            runOnUiThread(new Runnable() {
                @Override
                public void run() {
                    Bitmap bitmap = BitmapFactory.decodeByteArray(baos.toByteArray(), 0, baos.toByteArray().length);
                    img.setImageBitmap(bitmap);
                }
            });
        } catch (Exception e) {
            e.printStackTrace();
            Log.e("exception", e.toString());
        }
    }


}

File: AboutRecyclerView/src/main/java/com/example/myapplication3/MainActivity.kt
package com.example.myapplication3

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

//        resources.getStringArray(R.array.array)
        var list = resources.getStringArray(R.array.array)
        rv.adapter=FlowerAdapter(list)
    }
}

class FlowerAdapter(var list: Array<String>) : RecyclerView.Adapter<FlowerAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        var view = LayoutInflater.from(parent.context).inflate(R.layout.item, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bind(list[position])
    }

    override fun getItemCount(): Int {
        return list.size
    }


    /*
    * ViewHolder 负责存储RecyclerView中每一个单独的表项所需显式的信息。
    * RecyclerView只需要创建当前所显示的表项数量的ViewHolder外加存储中的几个ViewHolder即可
    * 随着用户的滑动屏幕，ViewHolder会被回收，（）使用新数据填充），已有的表项会在一端小时，并且在另一端显示一个新的表项
    * adapter从数据源获取数据，并将数据传递给正在更新其所持有试图的ViewHolder
    *
    * */
    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        //        constructor(itemView: View) : super(itemView)
        val tv: TextView = itemView.findViewById(R.id.tvItem)
        fun bind(text: String) {
            tv.text = text
        }
    }
}